### PR TITLE
Issu sw install

### PR DESCRIPTION
**Pull request by vnitinv**
function added _issu_nssu_requirement_validation

### CODE REVIEW

File: lib/jnpr/junos/utils/sw.py [hunk 1]
```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
...
+            with StartShell(self._dev) as ss:
+                ss.run('cli', '> ', timeout=5)
+                if ss.run('request routing-engine login other-routing-engine')[0]:
+                    # depending on user permission, prompt will go to either cli
+                    # or shell, below line of code prompt will finally end up in
+                    # cli mode
+                    ss.run('cli', '> ', timeout=5)
+                    data = ss.run('show system switchover', '> ', timeout=5)
+                    output = data[1]
+                    ss.run('exit')
```
Comment (stacywsmith) [inline lib/jnpr/junos/utils/sw.py]: Does this cleanly exit the `request routing-engine login` in all cases (both shell user and cli user)?

File: lib/jnpr/junos/utils/sw.py [hunk 2]
```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
...
+            * Check NSR is enabled
+            * Check commit synchronize is enabled
+            * Verify that NSR is configured on the master Routing Engine
+                by using the "show task replication" command.
+
+        :returns:
+            * ``True`` if validation passes.
+            * * ``False`` otherwise
+        """
         self.log('Checking GRES status')
```
Comment (stacywsmith) [inline lib/jnpr/junos/utils/sw.py]: Suggest changing to: `self.log('Checking GRES configuration')`

File: lib/jnpr/junos/utils/sw.py [hunk 3]
```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
...
         self.log('Checking commit synchronize status')
         conf = self._dev.rpc.get_config(
             filter_xml=etree.XML('<configuration><system><commit><synchronize/></commit></system></configuration>'),
-            options={'database': 'committed', 'inherit': 'inherit', 'commit-scripts': 'apply'})
+            options={'database': 'committed', 'inherit': 'inherit',
+                     'commit-scripts': 'apply'})
         if conf.find('system/commit/synchronize') is None:
             self.log('Requirement FAILED: commit synchronize is not Enabled in configuration')
             return False
+        self.log('Checking NSR status')
```
Comment (stacywsmith) [inline lib/jnpr/junos/utils/sw.py]: Suggest changing to: `self.log('Checking NSR configuration')`

File: lib/jnpr/junos/utils/sw.py [hunk 4]
```diff
@@ -270,44 +270,82 @@ def _issu_requirement_validation(self):
...
+        conf = self._dev.rpc.get_config(filter_xml=etree.XML(
+            '<configuration><routing-options><nonstop-routing/></routing-options></configuration>'),
+            options={'database': 'committed', 'inherit': 'inherit',
+                     'commit-scripts': 'apply'})
+        if conf.find('routing-options/nonstop-routing') is None:
+            self.log('Requirement FAILED: NSR is not Enabled in configuration')
+            return False
         self.log('Verifying that NSR is configured on the current Routing Engine\n'
```
Comment (stacywsmith) [inline lib/jnpr/junos/utils/sw.py]: Suggest changing to: `self.log('Verifying that GRES status on the current Routing Engine\n'`

File: lib/jnpr/junos/utils/sw.py [hunk 5]
```diff
@@ -248,11 +253,6 @@ def _issu_requirement_validation(self):
             * The master Routing Engine and backup Routing Engine must be
                 running the same software version before you can perform a
                 unified ISSU.
```
Comment (stacywsmith) [inline lib/jnpr/junos/utils/sw.py]: Add:
* All checks from _issu_nssu_requirement_validation()

File: lib/jnpr/junos/utils/sw.py (hunk 2 above)
Comment (vnitinv) [inline lib/jnpr/junos/utils/sw.py]: done

File: lib/jnpr/junos/utils/sw.py (hunk 3 above)
Comment (vnitinv) [inline lib/jnpr/junos/utils/sw.py]: done

File: lib/jnpr/junos/utils/sw.py (hunk 4 above)
Comment (vnitinv) [inline lib/jnpr/junos/utils/sw.py]: done

**Pull request merged**